web-client/extras/lib/SerialBuffer.ts

```diff
@@ -0,0 +1,205 @@
+    }
+
+    get writePos(): number {
+        return this._writePos;
+    }
+
+    set writePos(value: number) {
+        if (value < 0 || value > this.byteLength) throw `Invalid writePos ${value}`;
+        this._writePos = value;
+    }
+
+    /**
+     * Resets the read and write position of the buffer to zero.
+     */
+    reset(): void {
+        this._readPos = 0;
+        this._writePos = 0;
+    }
+
+    read(length: number): Uint8Array {
+        const value = this.subarray(this._readPos, this._readPos + length);
+        this._readPos += length;
+        return new Uint8Array(value);
+    }
+
+    write(array: Uint8Array): void {
+        this.set(array, this._writePos);
+        this._writePos += array.byteLength;
+    }
+
+    readUint8(): number {
+        return this._view.getUint8(this._readPos++);
+    }
+
+    writeUint8(value: number): void {
+        this._view.setUint8(this._writePos++, value);
+    }
+
+    readUint16(): number {
+        const value = this._view.getUint16(this._readPos);
+        this._readPos += 2;
+        return value;
+    }
+
+    writeUint16(value: number): void {
+        this._view.setUint16(this._writePos, value);
+        this._writePos += 2;
+    }
+
+    readUint32(): number {
+        const value = this._view.getUint32(this._readPos);
+        this._readPos += 4;
+        return value;
+    }
+
+    writeUint32(value: number): void {
+        this._view.setUint32(this._writePos, value);
+        this._writePos += 4;
+    }
+
+    readUint64(): number {
+        const value = this._view.getUint32(this._readPos) * Math.pow(2, 32) + this._view.getUint32(this._readPos + 4);
+        if (!NumberUtils.isUint64(value)) throw new Error('Malformed value');
+        this._readPos += 8;
+        return value;
+    }
+
+    writeUint64(value: number): void {
+        if (!NumberUtils.isUint64(value)) throw new Error('Malformed value');
+        this._view.setUint32(this._writePos, Math.floor(value / Math.pow(2, 32)));
+        this._view.setUint32(this._writePos + 4, value);
+        this._writePos += 8;
+    }
+
+    readVarUint(): number {
+        const value = this.readUint8();
+        if (value < 0xFD) {
+            return value;
+        } else if (value === 0xFD) {
+            return this.readUint16();
+        } else if (value === 0xFE) {
+            return this.readUint32();
+        } /*if (value === 0xFF)*/ else {
+            return this.readUint64();
+        }
+    }
+
+    writeVarUint(value: number): void {
+        if (!NumberUtils.isUint64(value)) throw new Error('Malformed value');
+        if (value < 0xFD) {
+            this.writeUint8(value);
+        } else if (value <= 0xFFFF) {
+            this.writeUint8(0xFD);
+            this.writeUint16(value);
+        } else if (value <= 0xFFFFFFFF) {
+            this.writeUint8(0xFE);
+            this.writeUint32(value);
+        } else {
+            this.writeUint8(0xFF);
+            this.writeUint64(value);
+        }
+    }
```

Comment:
I don't think we have these kinds of varuints in the code base anymore.

Reply:
Or is that a different _kind_ of varint?

Reply:
Different kind of varint: https://postcard.jamesmunns.com/wire-format#varint-encoded-integers.

Reply:
Done in https://github.com/nimiq/core-rs-albatross/pull/2715/commits/3103f08eaf93752c0376f93e25e2b079f81aaae1